dotnet/WPF/VisualLayerIntegration/VisualLayerIntegration/BarGraphUtility/BarBrushHelper.cs

```diff
@@ -30,111 +30,90 @@
 
         public BarBrushHelper(Compositor c)
         {
-            compositor = c;
-            rand = new Random();
+            _compositor = c;            
         }
 
-        internal CompositionBrush[] GenerateSingleColorBrush(int numBrushes, Color color)
+        internal CompositionBrush GenerateSingleColorBrush(Color color)
         {
-            var brushes = new CompositionBrush[numBrushes];
-            for (int i = 0; i < numBrushes; i++)
-            {
-                brushes[i] = compositor.CreateColorBrush(color);
-            }
-
-            return brushes;
+            return _compositor.CreateColorBrush(color);
         }
 
-
         internal CompositionBrush[] GenerateRandomColorBrushes(int numBrushes)
         {
-            var max = byte.MaxValue + 1; // 256
             var brushes = new CompositionBrush[numBrushes];
             for (int i = 0; i < numBrushes; i++)
             {
-                byte r = Convert.ToByte(rand.Next(max));
-                byte g = Convert.ToByte(rand.Next(max));
-                byte b = Convert.ToByte(rand.Next(max));
-                Color c = Color.FromArgb(Convert.ToByte(255), r, g, b);
-                brushes[i] = compositor.CreateColorBrush(c);
+                var rgb = new byte[3];
+                _rand.NextBytes(rgb);
+                var c = Color.FromArgb(255, rgb[0], rgb[1], rgb[2]);
+                brushes[i] = _compositor.CreateColorBrush(c);
             }
 
             return brushes;
         }
 
-        internal CompositionBrush[] GeneratePerBarLinearGradient(int numBrushes, List<Color> colors)
+        internal CompositionBrush GenerateLinearGradient(Color[] colors)
         {
-            var lgb = compositor.CreateLinearGradientBrush();
-            lgb.RotationAngleInDegrees = 45;
+            var linearGradientBrush = _compositor.CreateLinearGradientBrush();
+            linearGradientBrush.RotationAngleInDegrees = 45;
 
-            int i = 0;
+            var i = 0;
             foreach (Color color in colors)
             {
-                float offset = i / ((float)colors.Count - 1);
-                var stop = compositor.CreateColorGradientStop(offset, color);
+                var offset = i / ((float)colors.Length - 1);
+                var stop = _compositor.CreateColorGradientStop(offset, color);
 
                 i++;
-                lgb.ColorStops.Add(stop);
+                linearGradientBrush.ColorStops.Add(stop);
             }
 
-            var brushes = new CompositionBrush[numBrushes];
-            for (int j = 0; j < numBrushes; j++)
-            {
-                brushes[j] = lgb;
-            }
-            return brushes;
+            return linearGradientBrush;
         }
 
-        internal CompositionBrush[] GenerateAmbientAnimatingPerBarLinearGradient(int numBrushes, List<Color> colors)
+        internal CompositionBrush GenerateAmbientAnimatingLinearGradient(Color[] colors)
```

Comment:
>GenerateAmbientAnimatingLinearGradient [](start = 34, length = 38)

CreateAnimatedLinearGradientBrushes. AFAICT there is nothing Ambient about these.

Reply:
Ambient as in ambiently animating it's gradient

---
In reply to: [276440562](https://github.com/Microsoft/Windows.UI.Composition-Win32-Samples/pull/1#discussion_r276440562) [](ancestors = 276440562)